common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -172,8 +172,12 @@ def _get_modulestore_for_courseid(self, course_id=None):
             if mapping is not None:
                 return mapping
             else:
+                if isinstance(course_id, LibraryLocator):
```

Comment:
I see what you're saying now - sorry I misunderstood. There are many different `has_course` methods...

A library is not a course, though they have similar properties. A library ID (key) *is* a subclass of *CourseKey*, the most generic key for courses, but `LibraryLocator` is not a subclass of `CourseLocator` and the concept of a library is different from that of a course.

The reason I didn't put this code into `store.has_course` is that `store.get_course()` and `store.get_library()` are already separated, and `get_course()` will never return a library. So it doesn't make sense to me if I could call `store.has_course(library_id)` and get `True` but then call `store.get_course(library_id)` and get an exception.